add multitenant-managed-api installation type

MStokluska opened the pull request:
# Description
Add new install type "multitenant-managed-api"

# Verification
On BYOC cluster run:
- run `INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local` 
- run `INSTALLATION_TYPE=multitenant-managed-api make code/run`
- set the cluster storage to false
- Confirm that RHMI CR installation type is multitenant-managed-api and installation completes successfully. 
- Navigate to prometheus and confirm that only deadmans snitch alert is firing
- Navigate to Grafana and confirm that dashboards are available
- Make sure that 3scale and keycloak are available
- Uninstall by deleting RHMI CR
- Confirm that RHOAM + RHMI e2e tests are successful on this PR

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

austincunningham commented on config/configmap/cro-configmaps.yaml:
```diff
@@ -10,4 +10,6 @@ data:
     {"blobstorage":"openshift", "smtpcredentials":"openshift", "redis":"openshift", "postgres":"openshift"}
   managed-api: >
+    {"blobstorage":"openshift", "smtpcredentials":"openshift", "redis":"openshift", "postgres":"openshift"}
+  multitenant-managed-api: >
```
Not something for this pr or jira but at some stage we may want to update the sample config map in cro
https://github.com/integr8ly/cloud-resource-operator/blob/master/config/samples/cloud_resource_config.yaml

laurafitzgerald replied:
I think we want to keep the cro one integreatly agnostic so this change shouldn't impact the config map in cro.

austincunningham commented on pkg/resources/installTypesHelper.go:
```diff
@@ -0,0 +1,21 @@
+	return installType == "managed"
+}
+
+func IsRHOAMMultitenant(installType integreatlyv1alpha1.InstallationType) bool {
```
Not seeing this being used, Guess its because we don't have any multi tenant config in the operator yet

MStokluska replied:
Yes, it will be used by other functionalities provided

austincunningham commented on pkg/resources/installTypesHelper.go:
```diff
@@ -0,0 +1,21 @@
+}
+
+func isMultitenant(installType integreatlyv1alpha1.InstallationType) bool {
+	return installType == "multitenant-managed-api"
```
Not sure where we would use this ? wouldn't the previous function `IsRHOAMMultitenant` cover this already.

MStokluska replied:
`IsRHOAMMultitenant` requires `isMultitenant` function.

austincunningham replied:
Going blind thanks @MStokluska

laurafitzgerald commented on pkg/config/rhssoUser.go:
```diff
@@ -55,7 +55,7 @@ func (r *RHSSOUser) GetReplicasConfig(inst *integreatlyv1alpha1.RHMI) int {
 	}
 
-	if inst.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+	if inst.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) || inst.Spec.Type == string(integreatlyv1alpha1.InstallationTypeMultitenantManagedApi) {
```
Could you use the IsRHOAM function here?

MStokluska replied:
Good point, thanks Laura!

MStokluska replied:
Sorry, forgot that doing it gave me errors before. Can't do it due to https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/backup.go using pkg/config

austincunningham replied:
ah the old import cycle error https://stackoverflow.com/questions/28256923/import-cycle-not-allowed

austincunningham replied:
resources/backup imports the config package , so the config package can't import the resources package as it creates a Circular Dependency. moving installTypesHelper.go  to a new package would resolve the issue but not worth it to solve a single line of code

philbrookes replied:
It probably makes more sense to put these functions next to where the installTypes are defined? https://github.com/integr8ly/integreatly-operator/blob/c500b9c3ea76b23be11b3fb33005f6c587875221/apis/v1alpha1/rhmi_types.go#L44-L48

which may also address this import loop?

laurafitzgerald reviewed (commented):
One or two questions. Looking great. Tests are nice too. 👍

austincunningham commented:
@MStokluska seeing an error running cluster/prepare/local
```bash
INSTALLATION_TYPE=mulitenant-managed-api make cluster/prepare/local
Now using project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".

You can add applications to this project with the 'new-app' command. For example, try:

    oc new-app django-psql-example

to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:

    kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node

namespace/operator labeled
Already on project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".
/home/austincunningham/go/bin/kustomize build config/crd | oc apply -f -
customresourcedefinition.apiextensions.k8s.io/rhmiconfigs.integreatly.org created
customresourcedefinition.apiextensions.k8s.io/rhmis.integreatly.org created
secret/smtp created
secret/deadmanssnitch created
secret/pagerduty created
secret/addon-managed-api-service-parameters created
configmap/cloud-resources-aws-strategies created
serviceaccount/rhmi-operator created
Error: evalsymlink failure on 'config/rbac-' : lstat /home/austincunningham/repo-integr8ly/integreatly-operator/config/rbac-: no such file or directory
error: no objects passed to create
make: [Makefile:342: cluster/prepare/local] Error 1 (ignored)
```
don't see the same running `INSTALLATION_TYPE=managed-api make cluster/prepare/local`

makslion commented:
> @MStokluska seeing an error running cluster/prepare/local
> 
> ```shell
> INSTALLATION_TYPE=mulitenant-managed-api make cluster/prepare/local
> Now using project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".
> 
> You can add applications to this project with the 'new-app' command. For example, try:
> 
>     oc new-app django-psql-example
> 
> to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:
> 
>     kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node
> 
> namespace/operator labeled
> Already on project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".
> /home/austincunningham/go/bin/kustomize build config/crd | oc apply -f -
> customresourcedefinition.apiextensions.k8s.io/rhmiconfigs.integreatly.org created
> customresourcedefinition.apiextensions.k8s.io/rhmis.integreatly.org created
> secret/smtp created
> secret/deadmanssnitch created
> secret/pagerduty created
> secret/addon-managed-api-service-parameters created
> configmap/cloud-resources-aws-strategies created
> serviceaccount/rhmi-operator created
> Error: evalsymlink failure on 'config/rbac-' : lstat /home/austincunningham/repo-integr8ly/integreatly-operator/config/rbac-: no such file or directory
> error: no objects passed to create
> make: [Makefile:342: cluster/prepare/local] Error 1 (ignored)
> ```
> 
> don't see the same running `INSTALLATION_TYPE=managed-api make cluster/prepare/local`
> 
> think this is the offending line https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L329 or may be the line above

I had the same error yesterday. The reason for it - `INSTALLATION_SHORTHAND` variable in the makefile is not set. For me fix was is to pull a fresh copy of the branch :thinking:

austincunningham commented:
@makslion I pull this branch this morning

makslion commented:
While running reconcile loops I've spotted the following messages: 
```
INFO[2021-07-22T10:51:50+01:00] updating env variable to system app           envVarName=BACKEND_ROUTE envVarValue="{nil nil nil &SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:backend-listener,},Key:service_endpoint,Optional:nil,}}" foundVariable=true product=3scale

INFO[2021-07-22T10:58:50+01:00] updating env variable to system app           envVarName=BACKEND_PUBLIC_URL envVarValue="{nil nil nil &SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:backend-listener,},Key:route_endpoint,Optional:nil,}}" foundVariable=true product=3scale

INFO[2021-07-22T10:52:30+01:00] Looking for RHMI CR                           controller=namespacelabel_controller ns=
```
Are those expected? :thinking:

makslion commented:
Verified following: 
- `INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local` and then `INSTALLATION_TYPE=multitenant-managed-api make code/run` on the OSD cluster results in the `stage == complete` in RHMI CR 
- only the `DeadMansSwitch` alert is firing
- grafana dashboards are available, but spotted a downtime of the Grafana UI (not sure that this important) as well as "no data" for `redhat-rhoam-rhsso /Keycloak Metrics` and `redhat-rhoam-user-sso /Keycloak Metrics` 
![Screenshot from 2021-07-22 12-04-25](https://user-images.githubusercontent.com/37799801/126629678-d784bdbf-4a56-45a2-a445-7d9a744182b7.png)
- 3scale is available 
- keycloak (rhsso) is available, while for user-sso it says "Unexpected error when authenticating with identity provider" but that can be me messing something up 
- tests on this PR has succeded 

(further verification step - uninstallation is to be verified)

MStokluska commented:
> @MStokluska seeing an error running cluster/prepare/local
> 
> ```shell
> INSTALLATION_TYPE=mulitenant-managed-api make cluster/prepare/local
> Now using project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".
> 
> You can add applications to this project with the 'new-app' command. For example, try:
> 
>     oc new-app django-psql-example
> 
> to build a new example application in Python. Or use kubectl to deploy a simple Kubernetes application:
> 
>     kubectl create deployment hello-node --image=gcr.io/hello-minikube-zero-install/hello-node
> 
> namespace/operator labeled
> Already on project "operator" on server "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443".
> /home/austincunningham/go/bin/kustomize build config/crd | oc apply -f -
> customresourcedefinition.apiextensions.k8s.io/rhmiconfigs.integreatly.org created
> customresourcedefinition.apiextensions.k8s.io/rhmis.integreatly.org created
> secret/smtp created
> secret/deadmanssnitch created
> secret/pagerduty created
> secret/addon-managed-api-service-parameters created
> configmap/cloud-resources-aws-strategies created
> serviceaccount/rhmi-operator created
> Error: evalsymlink failure on 'config/rbac-' : lstat /home/austincunningham/repo-integr8ly/integreatly-operator/config/rbac-: no such file or directory
> error: no objects passed to create
> make: [Makefile:342: cluster/prepare/local] Error 1 (ignored)
> ```
> 
> don't see the same running `INSTALLATION_TYPE=managed-api make cluster/prepare/local`
> 
> think this is the offending line https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L329
> 
> went through the rest of the verification but seeing issues in the install, suspect due to the rbac failure during `make cluster/prepare/local`
> 
> ```
> E0722 11:32:13.891762  519839 reflector.go:178] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: Get "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443/apis/keycloak.org/v1alpha1/namespaces/redhat-rhoam-user-sso/keycloakrealms?resourceVersion=73166&timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
> I0722 11:32:14.812243  519839 trace.go:205] Trace[1237958226]: "Reflector ListAndWatch" name:sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136 (22-Jul-2021 11:32:04.811) (total time: 10000ms):
> Trace[1237958226]: [10.000903325s] [10.000903325s] END
> E0722 11:32:14.812258  519839 reflector.go:178] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Postgres: Get "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443/apis/integreatly.org/v1alpha1/namespaces/redhat-rhoam-cloud-resources/postgres?resourceVersion=61738&timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
> I0722 11:32:15.522969  519839 trace.go:205] Trace[1615891013]: "Reflector ListAndWatch" name:sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136 (22-Jul-2021 11:32:05.522) (total time: 10000ms):
> Trace[1615891013]: [10.000620213s] [10.000620213s] END
> E0722 11:32:15.522982  519839 reflector.go:178] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: Get "https://api.mw-collab-byoc.4lp9.s1.devshift.org:6443/apis/keycloak.org/v1alpha1/namespaces/redhat-rhoam-user-sso/keycloakclients?resourceVersion=65611&timeout=10s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
> ```
> 
> will clean the cluster and try again

The command you have is:
`INSTALLATION_TYPE=mulitenant-managed-api make cluster/prepare/local` it should be
`INSTALLATION_TYPE=multienant-managed-api make cluster/prepare/local` - typo in MULTITENANT vs yours MULITENANT

@austincunningham

MStokluska commented:
> Verified following:
> 
>     * `INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local` and then `INSTALLATION_TYPE=multitenant-managed-api make code/run` on the OSD cluster results in the `stage == complete` in RHMI CR
> 
>     * only the `DeadMansSwitch` alert is firing
> 
>     * grafana dashboards are available, but spotted a downtime of the Grafana UI (not sure that this important) as well as "no data" for `redhat-rhoam-rhsso /Keycloak Metrics` and `redhat-rhoam-user-sso /Keycloak Metrics`
>       ![Screenshot from 2021-07-22 12-04-25](https://user-images.githubusercontent.com/37799801/126629678-d784bdbf-4a56-45a2-a445-7d9a744182b7.png)
> 
>     * 3scale is available
> 
>     * keycloak (rhsso) is available, while for user-sso it says "Unexpected error when authenticating with identity provider" but that can be me messing something up
> 
>     * tests on this PR has succeded
> 
> 
> (further verification step - uninstallation is to be verified)

@makslion you need to select ns for that dashboard to work :)

austincunningham commented:
@MStokluska ah the verification steps have the typo, nothing every bad happens from copy and paste :)

MStokluska commented:
> While running reconcile loops I've spotted the following messages:
> 
> ```
> INFO[2021-07-22T10:51:50+01:00] updating env variable to system app           envVarName=BACKEND_ROUTE envVarValue="{nil nil nil &SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:backend-listener,},Key:service_endpoint,Optional:nil,}}" foundVariable=true product=3scale
> 
> INFO[2021-07-22T10:58:50+01:00] updating env variable to system app           envVarName=BACKEND_PUBLIC_URL envVarValue="{nil nil nil &SecretKeySelector{LocalObjectReference:LocalObjectReference{Name:backend-listener,},Key:route_endpoint,Optional:nil,}}" foundVariable=true product=3scale
> 
> INFO[2021-07-22T10:52:30+01:00] Looking for RHMI CR                           controller=namespacelabel_controller ns=
> ```
> 
> Are those expected? thinking

I presume so, there was no changes done that would affect it, I presume it is same on master.

MStokluska commented:
> @MStokluska ah the verification steps have the typo, nothing bad every happens from copy and paste :)

Updated the description, thanks!

austincunningham commented:
- installed locally 
- installation completed
- rhmi cr using multitenant type
- rhmi cr useClusterStorage: 'false'
```yaml
spec:
  smtpSecret: redhat-rhoam-smtp
  selfSignedCerts: true
  useClusterStorage: 'false'
  pagerDutySecret: redhat-rhoam-pagerduty
  alertingEmailAddresses:
    businessUnit: noreply-test@rhmi-redhat.com
    cssre: noreply-test@rhmi-redhat.com
  masterURL: console-openshift-console.apps.mw-collab-byoc.4lp9.s1.devshift.org
  namespacePrefix: redhat-rhoam-
  pullSecret:
    name: ''
    namespace: ''
  deadMansSnitchSecret: redhat-rhoam-deadmanssnitch
  type: multitenant-managed-api
  routingSubdomain: apps.mw-collab-byoc.4lp9.s1.devshift.org
  priorityClassName: rhoam-pod-priority
```
- no alerts firing except deadman
- Grafana step a bit unclear, 
  - customer grafana available
  - redhat-rhoam-middleware-monitoring-operator dashboards available
  - Like @makslion see no data on both the keycloak dashboards.(not sure if this has any bearing on this change but would guess not)
- usersso available and 3scale avaliable
- deleting the rhmi cr deletes rhoam
- e2e tests pass

austincunningham commented:
/lgtm

philbrookes commented on Makefile:
```diff
@@ -85,6 +85,19 @@ ifeq ($(INSTALLATION_TYPE), managed-api)
 	INSTALLATION_SHORTHAND ?= rhoam
 endif
 
+ifeq ($(INSTALLATION_TYPE), multitenant-managed-api)
```
I wonder if we should define this in separate files, and name them after the install type, and can we just include them then? 
e.g.
`multitenant-managed-api.env`:
```
PROJECT=managed-api-service
TAG ?= $(RHOAM_TAG)
OPERATOR_IMAGE=$(REG)/$(ORG)/$(PROJECT):v$(TAG)
NAMESPACE_PREFIX ?= redhat-rhoam-
APPLICATION_REPO ?= managed-api-service
INSTALLATION_PREFIX ?= redhat-rhoam
OLM_TYPE ?= multitenant-managed-api-service
INSTALLATION_NAME ?= rhoam
INSTALLATION_SHORTHAND ?= rhoam
```
and replace the below with:
```
include $(INSTALLATION_TYPE).env
```
WDYT?

MStokluska replied:
I really like this idea, updated it now.

philbrookes commented on pkg/config/threescale.go:
```diff
@@ -118,7 +118,7 @@ func (t *ThreeScale) GetReplicasConfig(inst *integreatlyv1alpha1.RHMI) map[strin
 	if resources.RunningInProw(inst) {
 		setDefaultNumberOfReplicas(1, threeScaleComponents)
-	} else if inst.Spec.Type != string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+	} else if inst.Spec.Type != string(integreatlyv1alpha1.InstallationTypeManagedApi) && inst.Spec.Type != string(integreatlyv1alpha1.InstallationTypeMultitenantManagedApi) {
```
moving them, may also allow them to be used here. Really would like to avoid the same logic implemented in different ways in different places. Seems like a prime area to cause bugs when we've forgotten these few exceptions next year!

MStokluska replied:
Thanks, good idea, I've moved it.

MStokluska commented:
@austincunningham @laurafitzgerald  @makslion I have pushed some changes based on the comments: 
- updated makefile by creating new env files that contain the envs required for each installation type.
- moved the Installation Type helper functions to apis/v1alpha1 and updated all the changes where they were used

Going to run e2e and rhoam-e2e again on this PR just to make sure everything is still in order.

MStokluska commented:
/test rhoam-e2e
/test e2e

MStokluska commented:
Repeating e2e due to "Cluster initialization failed"
/test e2e

KevFan commented on apis/v1alpha1/installationTypesHelper.go:
```diff
@@ -0,0 +1,17 @@
+package v1alpha1
+
+func IsRHOAM(installType InstallationType) bool {
+	return installType == "managed-api" || installType == "multitenant-managed-api"
+}
+
+func IsRHMI(installType InstallationType) bool {
+	return installType == "managed"
+}
+
+func IsRHOAMMultitenant(installType InstallationType) bool {
+	return IsRHOAM(installType) && isMultitenant(installType)
+}
+
+func isMultitenant(installType InstallationType) bool {
+	return installType == "multitenant-managed-api"
```
```suggestion
func IsRHOAM(installType InstallationType) bool {
	return installType == InstallationTypeManagedApi ||  isMultitenant(installType)
}

func IsRHMI(installType InstallationType) bool {
	return installType == InstallationTypeManaged
}

func IsRHOAMMultitenant(installType InstallationType) bool {
	return IsRHOAM(installType) && isMultitenant(installType)
}

func isMultitenant(installType InstallationType) bool {
	return installType == InstallationTypeMultitenantManagedApi
```

KevFan commented on apis/v1alpha1/installationTypesHelper_test.go:
```diff
@@ -0,0 +1,113 @@
+	}{
+		{
+			name:            "test that isRHMI returns true",
+			installType:     "managed",
+			expectedOutcome: true,
+		},
+		{
+			name:            "test that isRHMI returns false",
+			installType:     "managed-api",
```
```suggestion
			installType:     InstallationTypeManaged,
			expectedOutcome: true,
		},
		{
			name:            "test that isRHMI returns false",
			installType:     InstallationTypeManagedApi,
```

KevFan commented on apis/v1alpha1/installationTypesHelper_test.go:
```diff
@@ -0,0 +1,113 @@
+		expectedOutcome bool
+	}{
+		{
+			name:            "test that isRHOAM returns true",
+			installType:     "managed-api",
+			expectedOutcome: true,
+		},
+		{
+			name:            "test that isRHOAM returns false",
+			installType:     "managed",
```
```suggestion
			name:            "test that isRHOAM returns true",
			installType:     InstallationTypeManagedApi,
			expectedOutcome: true,
		},
		{
			name:            "test that isRHOAM returns false",
			installType:     InstallationTypeManaged,
```

KevFan commented on apis/v1alpha1/installationTypesHelper_test.go:
```diff
@@ -0,0 +1,113 @@
+		expectedOutcome bool
+	}{
+		{
+			name:            "test that isMultitenant returns true",
+			installType:     "multitenant-managed-api",
+			expectedOutcome: true,
+		},
+		{
+			name:            "test that isMultitenant returns false",
+			installType:     "managed",
+			expectedOutcome: false,
```
```suggestion
			name:            "test that isMultitenant returns true",
			installType:     InstallationTypeMultitenantManagedApi,
			expectedOutcome: true,
		},
		{
			name:            "test that isMultitenant returns false",
			installType:     InstallationTypeManaged,
			expectedOutcome: false,
```

KevFan commented on apis/v1alpha1/installationTypesHelper_test.go:
```diff
@@ -0,0 +1,113 @@
+		expectedOutcome bool
+	}{
+		{
+			name:            "test that isMultitenant returns true",
+			installType:     "multitenant-managed-api",
+			expectedOutcome: true,
+		},
+		{
+			name:            "test that isMultitenant returns false",
+			installType:     "managed",
+			expectedOutcome: false,
```
```suggestion
			name:            "test that isMultitenant returns true",
			installType:    InstallationTypeMultitenantManagedApi,
			expectedOutcome: true,
		},
		{
			name:            "test that isMultitenant returns false",
			installType:     InstallationTypeManaged,
			expectedOutcome: false,
```

KevFan commented on controllers/rhmi/rhmi_controller.go:
```diff
@@ -839,7 +839,7 @@ func (r *RHMIReconciler) preflightChecks(installation *rhmiv1alpha1.RHMI, instal
 	}
 
-	if installation.Spec.Type == string(rhmiv1alpha1.InstallationTypeManaged) || installation.Spec.Type == string(rhmiv1alpha1.InstallationTypeManagedApi) {
+	if rhmiv1alpha1.IsRHOAM(rhmiv1alpha1.InstallationType(installation.Spec.Type)) || rhmiv1alpha1.IsRHMI(rhmiv1alpha1.InstallationType(installation.Spec.Type)) {
```
Wonder would it be good to add and use a `isManaged` fucntion for this instead while we're these `isX` functions ? :thinking:

MStokluska replied:
Thanks @KevFan, I've addressed your comments, thanks a lot for suggestions :)
I've also added the IsManaged to cover RHMI/RHOAM/MtRHOAM. Thanks!

KevFan reviewed (commented):
Looks good ! :+1: 

Just have some very minor suggestions / comments

MStokluska commented:
/test e2e

MStokluska commented:
/test rhoam-e2e

austincunningham commented:
## Verification
repeated the verification
- installed locally 
- installation completed
```yaml
    products:
      name: products
      phase: completed
```
- rhmi cr using multitenant type
- rhmi cr useClusterStorage: 'false'
```yaml
spec:
  smtpSecret: redhat-rhoam-smtp
  selfSignedCerts: true
  useClusterStorage: 'false'
  pagerDutySecret: redhat-rhoam-pagerduty
  alertingEmailAddresses:
    businessUnit: noreply-test@rhmi-redhat.com
    cssre: noreply-test@rhmi-redhat.com
  masterURL: console-openshift-console.apps.mw-collab-byoc.5tqa.s1.devshift.org
  namespacePrefix: redhat-rhoam-
  pullSecret:
    name: ''
    namespace: ''
  deadMansSnitchSecret: redhat-rhoam-deadmanssnitch
  type: multitenant-managed-api
  routingSubdomain: apps.mw-collab-byoc.5tqa.s1.devshift.org
  priorityClassName: rhoam-pod-priority
```
- no alerts firing except deadman
![image](https://user-images.githubusercontent.com/16667688/126976966-c17a10b0-8314-49ab-9211-fd69476337f4.png)

- Grafana step a bit unclear, 
  - customer grafana available
  - redhat-rhoam-middleware-monitoring-operator dashboards available

- usersso available and 3scale avaliable
- deleting the rhmi cr deletes rhoam

austincunningham commented:
/lgtm
